lib/langchain/tool/database.rb

```diff
@@ -0,0 +1,37 @@
+module Langchain::Tool
+  class Database < Base
+    description <<~DESC
+      Useful for getting the result of a database query.
+
+      The input to this tool should be valid SQL.
+    DESC
+
+    # Establish a database connection
+    # example: 'postgres://user:password@localhost:5432/db_name'
+    def initialize(db_connection_string)
+      depends_on "sequel"
+      require "sequel"
+      require "sequel/extensions/schema_dumper"
+
+      raise StandardError, "db_connection_string parameter cannot be blank" if db_connection_string.empty?
+
+      @db = Sequel.connect(db_connection_string)
+      @db.extension :schema_dumper
+    end
+
+    def schema
+      Langchain.logger.info("[#{self.class.name}]".light_blue + ": Dumping schema")
+      @db.dump_schema_migration(same_db: true, indexes: false) unless @db.adapter_scheme == :mock
+    end
+
+    # Evaluates a sql expression and outputs labeled values
+    # @param input [String] sql expression
+    # @return [String] results
+    def execute(input:)
+      Langchain.logger.info("[#{self.class.name}]".light_blue + ": Executing \"#{input}\"")
+      @db[input].map do |row|
+        row.map { |key, value| key.to_s + ": " + value.to_s }.each { |col| col.to_s }.join(", ")
```

Comment:
Would it be better to just return `@db[input].to_a` here?
```ruby
[1] pry(#<Langchain::Tool::Database>)> @db[input].to_a
=> [{:id=>8, ... }]
```

Reply:
That usually works, but the ai seemed to work better yesterday with multiple row return sets, so I'm thinking a table format with a header might be best.
And some queries seem to be getting off and we get a messy stack trace when the query fails. Maybe a simple message is better?  

```
[LangChain.rb] [Langchain::Tool::Database]: Executing "
SELECT id, name, email, instance_role_id, image_"
/Users/mattlindsey/.rvm/gems/ruby-3.2.1/gems/sequel-5.68.0/lib/sequel/adapters/postgres.rb:171:in `exec': PG::UndefinedColumn: ERROR:  column "id" does not exist (Sequel::DatabaseError)
LINE 2: SELECT id, name, email, instance_role_id, image_
               ^

	from /Users/mattlindsey/.rvm/gems/ruby-3.2.1/gems/sequel-5.68.0/lib/sequel/adapters/postgres.rb:171:in `block in execute_query'
	from /Users/mattlindsey/.rvm/gems/ruby-3.2.1/gems/sequel-5.68.0/lib/sequel/database/logging.rb:38:in `log_connection_yield'
	from /Users/mattlindsey/.rvm/gems/ruby-3.2.1/gems/sequel-5.68.0/lib/sequel/adapters/postgres.rb:171:in `execute_query'
	...

Reply:
I added more graceful handling.